cli/run.go

```diff
@@ -0,0 +1,139 @@
+		Use:   "run a container",
+		Short: "Create a new container and start it",
+		Long:  runDescription,
+		Args:  cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return rc.runRun(args)
+		},
+		Example: runExample(),
+	}
+	rc.addFlags()
+}
+
+// addFlags adds flags for specific command.
+func (rc *RunCommand) addFlags() {
+	flagSet := rc.cmd.Flags()
+	flagSet.StringVar(&rc.name, "name", "", "Specify name of container")
+	flagSet.BoolVarP(&rc.tty, "tty", "t", false, "Allocate a tty device")
+	flagSet.StringSliceVarP(&rc.volume, "volume", "v", nil, "Bind mount volumes to container")
+	flagSet.StringVar(&rc.runtime, "runtime", "", "Specify oci runtime")
+	flagSet.BoolVarP(&rc.attach, "attach", "a", false, "Attach container's STDOUT and STDERR")
+	flagSet.BoolVarP(&rc.stdin, "interactive", "i", false, "Attach container's STDIN")
+	flagSet.BoolVarP(&rc.rm, "rm", "", false, "Remove the container after running")
+}
+
+// runRun is the entry of run command.
+func (rc *RunCommand) runRun(args []string) error {
+	config := rc.config()
+
+	ref, err := reference.Parse(args[0])
+	if err != nil {
+		return fmt.Errorf("failed to run container: %v", err)
+	}
+
+	config.Image = ref.String()
+	if len(args) == 2 {
+		config.Cmd = strings.Fields(args[1])
+	}
+	containerName := rc.name
+
+	apiClient := rc.cli.Client()
+	result, err := apiClient.ContainerCreate(config.ContainerConfig, config.HostConfig, containerName)
+	if err != nil {
+		return fmt.Errorf("failed to run container: %v", err)
+	}
+	if len(result.Warnings) != 0 {
+		fmt.Printf("WARNING: %s \n", strings.Join(result.Warnings, "\n"))
+	}
+
+	// pouch run not specify --name
+	if containerName == "" {
+		containerName = result.Name
+	}
+
+	if rc.rm {
+		defer func() {
+			if err := apiClient.ContainerRemove(containerName, false); err != nil {
+				fmt.Printf("%v\n", err)
+			}
+		}()
+	}
+
+	var wait chan struct{}
+	if rc.attach || rc.stdin {
+		in, out, err := setRawMode(rc.stdin, false)
+		if err != nil {
+			return fmt.Errorf("failed to set raw mode")
+		}
+		defer func() {
+			if err := restoreMode(in, out); err != nil {
+				fmt.Fprintf(os.Stderr, "failed to restore term mode")
+			}
+		}()
+
+		conn, br, err := apiClient.ContainerAttach(containerName, rc.stdin)
+		if err != nil {
+			return fmt.Errorf("failed to attach container: %v", err)
+		}
+
+		wait = make(chan struct{})
+		go func() {
+			io.Copy(os.Stdout, br)
+			close(wait)
+		}()
+		go func() {
+			io.Copy(conn, os.Stdin)
+			close(wait)
+		}()
+	}
+
+	// start container
+	if err := apiClient.ContainerStart(containerName, ""); err != nil {
+		return fmt.Errorf("failed to run container %s: %v", containerName, err)
+	}
+
+	// wait the io to finish
+	if rc.attach || rc.stdin {
+		<-wait
+	}
+	return nil
+}
+
+// runExample shows examples in run command, and is used in auto-generated cli docs.
+func runExample() string {
+	return `$ pouch run --name test registry.hub.docker.com/library/busybox:latest
```

Comment:
Do we need to output something if one makes it to run a container?
Currently I see no output, right?

Reply:
Yeah, we should print Container ID when not specify -a and -i